src/ProductConstructionService/ProductConstructionService.Api/VirtualMonoRepo/VmrConfiguration.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Maestro.DataProviders;
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace ProductConstructionService.Api.VirtualMonoRepo;
+
+public static class VmrConfiguration
+{
+    public const string VmrPathKey = "VmrPath";
+    public const string TmpPathKey = "TmpPath";
+    public const string VmrUriKey = "VmrUri";
+
+    public static void AddVmrRegistrations(this WebApplicationBuilder builder)
+    {
+        builder.Services.TryAddSingleton<IBasicBarClient, SqlBarClient>();
+        builder.Services.AddVmrManagers(
+            "git",
+            Environment.GetEnvironmentVariable(VmrPathKey) ?? throw new ArgumentException($"{VmrPathKey} environmental variable must be set"),
+            Environment.GetEnvironmentVariable(TmpPathKey) ?? throw new ArgumentException($"{TmpPathKey} environmental variable must be set"),
```

Comment:
It would be better to have these as parameters of the `Add` method as this class could be used from other scenarios where I have other ways to inject these. And we'd read the env in the `Program.cs`. I think it will be more discoverable there.

Reply:
sure, moved